java-client/src/main/java/co/elastic/clients/transport/rest_client/RestClientOptions.java

```diff
@@ -63,7 +65,8 @@ static RestClientOptions of(@Nullable TransportOptions options) {
         return builder.build();
     }
 
-    public RestClientOptions(RequestOptions options) {
+    public RestClientOptions(RequestOptions options, boolean keepResponseBodyOnException) {
```

Comment:
We should keep the existing constructor without the additional parameter to avoid a breaking change (and default the additional parameter to `false`).

Also, to be future proof if we add more options in the features, what about changing this constructor to `RestClientOptions(RequestOptions.Builder builder)` so that we can add more properties without impacting the constructor's signature?

Reply:
I think the future proof constructor will make more sense when we'll have control over RequestOptions in the low level client!